scala-thoughts.md

```diff
@@ -0,0 +1,51 @@
+#Scala Thoughts#
+
+###What's easy to do in Scala?  What's not?###
+
+Functional programming seems relatively easy in Scala.  In this respect, the
+language had many similarities to other functional languages like Racket and
+Haskell.  I didn't explicitly have to say "return", as is true of other functional
+languages.  The use of higher-order functions was similar to my experiences with
+Racket as well.  Scala also does support pattern matching like Haskell, but it
+seemed slightly more cumbersome.  This made it very easy to, for example, return
+different results based on different inputs, as we did in the `matchOnInputType`
+function.  It was also very easy to declare variables and use type inference, at
+least in the scope of this assignment.  Also, the collections seemed to have
+extensive and intuitives APIs, supported by thorough documentation online.  This
+made it very easy to take advantage of the built-in functionality of lists.
+
+###What are your favorite choices that the designers of Scala made? Why?###
+
+There were a few design choices that I very much appreciated.  I appreciated
+the different ways you could use pattern matching.  In match/case statements,
+it was possible to include a line like `case x : Int if x > 0`, so that we
+could match by type and a predicate.  I cannot think of ways to express
+something as concisely in other languages.  I also thought that the syntax was
+relatively intuitive.  I appreciated that we didn't have to use lots of nested
+parentheses as in Racket or LISP.  It was possible to declare new variables with
+val (as if we were using let), and then return a result.  I was also impressed
+by the testing frameworks.  It seems like Scala is very customizable, allowing
+programmers to use syntax like `it should "do something" in {...}`.  I am
+excited to learn more about these options, as they seem very useful for creating
+a DSL.
+
+###What are your least-favorite design choices?###
+
+There were a couple of small situations that I thought were simpler or more
+intuitive in other languages.  For example, if `myList` is a list, I am more
+familiar with using `myList[0]` instead of `myList(0)` to access the first
+element.  I am unsure why the language designers chose this, perhaps because the
+bracket operator has a different use.  I also thought that the syntax for naming
+a variable was unusual.  I don't understand why `val x: Int = 5` is better than,
```

Comment:
I definitely agree with this. Personally, it's also very confusing that val and var are one letter apart, so I could definitely see typos causing bugs in the code. I agree that `const` seems better and would be curious to hear the designers' reasoning.